src/BodyParams/JsonStrategy.php

```diff
@@ -27,9 +27,7 @@ class JsonStrategy implements StrategyInterface
 {
     public function match(string $contentType) : bool
     {
-        $parts = explode(';', $contentType);
-        $mime = array_shift($parts);
-        return (bool) preg_match('#[/+]json$#', trim($mime));
+        return (bool) preg_match('#^application/(|[\s\S]+\+)json($|[ ;])#', $contentType);
```

Comment:
Why is `\s` in the character set? Whitespace between the `/` and `json` would be invalid. I'm thinking the following makes more sense:

```php
return (bool) preg_match('#^application/(|\S+\+)json($|[ ;])#', $contentType);
```

Reply:
I will add test case